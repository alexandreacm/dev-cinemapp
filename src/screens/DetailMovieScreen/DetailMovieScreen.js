import React, { useEffect, useMemo } from 'react';
import styled from 'styled-components/native';

import {
  useRoute,
  useIsFocused,
  useNavigation
} from '@react-navigation/native';
import { useSelector, useDispatch } from 'react-redux';

import colors from '@/theme/colors';
import Header from '@/components/Header';
import CardDetailMovie from '@/components/CardDetailMovie';

import { DETAILS_MOVIE } from '@/store/slices/movieSlice';

const DetailsMovieScreen = () => {
  const isFocused = useIsFocused();
  const { goBack } = useNavigation();
  const dispatch = useDispatch();

  const { isLoading, movieDetailsData, errorMessage } = useSelector(
    ({ movies }) => movies
  );
  const {
    params: { id }
  } = useRoute();

  const handleGoBack = () => goBack();

  const contentDetailComic = useMemo(() => {
    if (isLoading)
      return <StyledActivityIndicator color={colors.PRIMARY} size='large' />;

    return (
      <CardDetailMovie
        key={movieDetailsData?.imdbID}
        poster={movieDetailsData?.Poster}
        title={movieDetailsData?.Title}
        description={
          movieDetailsData?.Writer ? movieDetailsData?.Writer : 'Sem descrição'
        }
        imdbRating={movieDetailsData?.imdbRating}
        year={movieDetailsData?.Year}
        runTime={movieDetailsData?.Runtime}
        actors={movieDetailsData?.Actors}
      />
    );
  }, [isLoading, movieDetailsData]);

  useEffect(() => {
    dispatch(DETAILS_MOVIE({ id }));
  }, [dispatch, id]);

  return (
    <>
      <Header
        showBackButton
        leftIcon
        onBackPress={handleGoBack}
        backgroundColor={colors.PRIMARY}
        backButtonColor={colors.COLOR_WHITE}
        isFocuse={isFocused}
      />
      {errorMessage && (
        <StyledViewLabel>
          <StyledLabel>{errorMessage}</StyledLabel>
        </StyledViewLabel>
      )}
      <StyledScrollView showsVerticalScrollIndicator={false}>
        <StyledContainerDefault>{contentDetailComic}</StyledContainerDefault>
      </StyledScrollView>
    </>
  );
};

const StyledContainerDefault = styled.View`
  flex: 1;
  background-color: ${colors.COLOR_WHITE};
  margin-left: 6px;
  margin-right: 6px;
  margin-top: 10px;
`;

const StyledLabel = styled.Text`
  width: 100%;
  font-weight: 400;
  font-size: 14px;
  margin-bottom: 8px;
  color: ${colors.PRIMARY};
`;

const StyledViewLabel = styled.View`
  width: 100%;
  padding: 10px;
  align-items: center;
  background-color: ${colors.DANGER};
`;

const StyledScrollView = styled.ScrollView`
  flex: 1;
  background: ${colors.COLOR_WHITE};
`;

const StyledActivityIndicator = styled.ActivityIndicator`
  height: 100px;
  margin-top: 32px;
  margin-bottom: 32px;
`;

export default DetailsMovieScreen;
