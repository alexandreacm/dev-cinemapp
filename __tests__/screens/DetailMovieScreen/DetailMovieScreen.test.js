import 'react-native';
import React from 'react';
import {
  matchSnapshotWithTheme,
  renderWithTheme
} from '@/helpers/functions/testing';

import DetailMovieScreen from '@/screens/DetailMovieScreen';

describe('tests in DetailMovieScreen Screen', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(<DetailMovieScreen />);
  });

  test('renders correctly', () => {
    renderWithTheme(<DetailMovieScreen />);
  });
});
