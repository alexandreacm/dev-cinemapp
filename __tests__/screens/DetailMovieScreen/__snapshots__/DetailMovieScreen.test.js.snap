// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in DetailMovieScreen Screen matches snapshot 1`] = `
Array [
  <RCTSafeAreaView
    backgroundColor="#25265E"
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#25265E",
        },
      ]
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "height": 20,
        },
        Object {
          "backgroundColor": "#25265E",
        },
      ]
    }
  />,
  <View
    backgroundColor="#25265E"
    slim={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#25265E",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 26,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 37,
            "justifyContent": "center",
            "marginRight": 27,
            "opacity": 1,
            "width": 37,
          }
        }
        testID="left-icon"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/icons/icMessage.png",
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 37,
            "justifyContent": "center",
            "marginRight": 27,
            "opacity": 1,
            "width": 37,
          }
        }
        testID="right-icon"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/icons/icBell.png",
            }
          }
        />
      </View>
    </View>
  </View>,
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFF",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginLeft": 6,
              "marginRight": 6,
              "marginTop": 10,
            },
          ]
        }
      >
        <View
          onPress={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#FFF",
                "borderRadius": 8,
                "elevation": 10,
                "marginBottom": 10,
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
                "shadowColor": "rgba(0, 0, 0, 0.15)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 25,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                  "width": "100%",
                },
              ]
            }
          >
            <Image
              resizeMode="cover"
              source={
                Object {
                  "uri": "",
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                    "height": 150,
                    "marginRight": 10,
                    "width": 100,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFF",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 16,
                      "fontWeight": "bold",
                      "marginBottom": 6,
                      "textAlign": "left",
                    },
                  ]
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#878787",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Rating: 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#878787",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Ano: 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#878787",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Tempo de filme: 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#878787",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "marginTop": 10,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Atores: 
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#f3f3f3",
                  "marginBottom": 8,
                  "marginTop": 10,
                  "paddingBottom": 5,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                  "paddingTop": 5,
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Summary
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "textAlign": "left",
                },
              ]
            }
          >
            Sem descrição
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
